src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -207,32 +207,43 @@ string EnsureRequiredTagIsPresent(DicomTag dicomTag)
         StoreValidationResultBuilder validationResultBuilder)
     {
         IReadOnlyCollection<QueryTag> queryTags = await _queryTagService.GetQueryTagsAsync();
-        foreach (DicomElement de in dicomDataset)
+        foreach (DicomItem item in dicomDataset)
         {
             try
             {
-                string value = dicomDataset.GetString(de.Tag);
-                if (value != null && value.EndsWith('\0'))
+                if (item.ValueRepresentation == DicomVR.SQ)
```

Comment:
While not likely, it could be that we've been given a DICOM dataset with more than expected depth, so I think we have to make this iterative. Although I like the use of recursion, and it makes the intent quite clear